src/pages/Details.js

```diff
@@ -9,6 +10,17 @@ import Navigation from '../components/Navigation/Navigation';
 export default function Details({menusdata}) {
   const params = useParams();
   const currentMenu = menusdata?.find(menu => menu.id === params.id);
+  let actives = [false, false, false, false, false];
+
+  function setActive() {
+    for (let i = 0; i < currentMenu.spiciness; i++) {
+      if (currentMenu.spiciness < actives.length) {
+        actives[i] = true;
+      }
+    }
+  }
+  setActive();
```

Comment:
I think "active" is a too general name and setActive is reserved for state setters. Additionally, I would integrate the array into the function, like so:

```suggestion
 let activePeppers = calculateActivePeppers();

  function calculateActivePeppers() {
    let pepperStatus = [false, false, false, false, false];
      for (let i = 0; i < currentMenu.spiciness; i++) {
        if (currentMenu.spiciness < pepperStatus.length) {
          pepperStatus[i] = true;
        }
      }
      return pepperStatus;
    }
```